testing/testDiamond.m

```diff
@@ -0,0 +1,118 @@
+function [success,blastStructure]=testDiamond(fullCheck)
+% testDiamond
+%   Performs a check for DIAMOND functionality in RAVEN. Depending on the
+%   parameter settings the user can choose between a quick check for
+%   binaries or the thorough testing while building DIAMOND database and
+%   running homology search with DIAMOND
+%
+%   Input:
+%   fullCheck       true if the thorough DIAMOND testing should be
+%                   performed (opt, default true)
+%
+%   Output: 
+%   success         true if the test was successful, otherwise equal to
+%                   zero
+%   blastStructure	blastStructure resulting from the thorough BLAST+ check
+%
+%   NOTE: The purpose of the thorough check is to assess whether the
+%   homology search can be successfully performed using existing BLAST+
+%   binaries. This testing function is completely standalone, only
+%   requiring DIAMOND binary and multi-FASTA files sce.fa and
+%   Sco_all_protein.faa from tutorials directory
+%
+%   Usage: [success,blastStructure]=testDiamond(fullCheck)
+%
+%   Simonas Marcisauskas, 2019-10-04
+%
+
+if nargin<1
+    fullCheck=true;
+end
+
+%Get the directory for RAVEN Toolbox
+[ST, I]=dbstack('-completenames');
+ravenPath=fileparts(fileparts(ST(I).file));
+
+if isunix
+    if ismac
+        binEnd='.mac';
+    else
+        binEnd='';
+    end
+elseif ispc
+    binEnd='';
+else
+    dispEM('Unknown OS, exiting.')
+    return
+end
+
+%Create an empty blastStructure. Even if a quick DIAMOND evaluation is
+%considered, blastStructure should still be in the output
+blastStructure=[];
+
+if ~fullCheck
+    fprintf(['Checking diamond' binEnd '... ']);
+    [res,~]=system(['"' fullfile(ravenPath,'software','diamond',['diamond' binEnd]) '"']);
+    if res==1
+        fprintf('OK\n');
+    else
+        fprintf('Not OK! The binary must be recompiled from source before running RAVEN\n');
+    end
+else    
+    %Generate temporary names for working directory and outFile
+    tmpDB=tempname;
+    outFile=tempname;
+    
+    %Run DIAMOND multi-threaded to use all logical cores assigned to MATLAB.
+    cores = evalc('feature(''numcores'')');
+    cores = strsplit(cores, 'MATLAB was assigned: ');
+    cores = regexp(cores{2},'^\d*','match');
+    cores = cores{1};
+    
+    %Create a temporary folder and copy both multi-FASTA files there
+    [~, ~]=system(['mkdir "' tmpDB '"']);
+    copyfile(fullfile(ravenPath,'tutorial','sce.fa'),tmpDB);
+    copyfile(fullfile(ravenPath,'tutorial','Sco_all_protein.faa'),tmpDB);
```

Comment:
Only one FASTA file is enough for the testing, the smaller one.

Reply:
Okay, will fix this soon